<?xml version="1.0"?>
<robot name="arm" version="1.0" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- Gazebo ROS Control plugin -->
  <!-- <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/arm</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo> -->

  <xacro:include filename="$(find sim)/urdf/arm.gazebo" />
  <xacro:include filename="$(find sim)/urdf/materials.xacro" />

  <!-- Fixing robot to base -->
  <link name="world"/>
  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="RoboticARM_-_Base-2_Base_bottom-1_link"/>
  </joint>

  <!-- Joints -->
  <joint name="b_joint" type="revolute">
    <limit lower="-3.14000" upper="3.14000" effort="10.0" velocity="1.0"/> <!--  -->
    <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00177 0.00000 0.72061"/>
    <parent link="RoboticARM_-_Base-2_Base_bottom-1_link"/>
    <child link="RoboticARM_-_Link1-1_link"/>
    <axis xyz="1.00000 0.00000 0.00000"/>
    <dynamics damping="0.7"/>
  </joint>
  <joint name="m_joint" type="revolute" >
    <limit lower="-3.14000" upper="3.14000" effort="10.0" velocity="1.0"/>
    <origin rpy="0.00000 0.00000 0.00000" xyz="0.00153 0.01211 1.09993"/>
    <parent link="RoboticARM_-_Link1-1_link"/>
    <child link="RoboticARM_-_Link2-1_link"/>
    <axis xyz="1.00000 0.00000 0.00000"/>
    <dynamics damping="0.7"/>
  </joint>
  <joint name="t_joint" type="revolute" >
    <limit lower="-3.14000" upper="3.14000" effort="10.0" velocity="1.0"/>
    <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00224 0.00221 0.85010"/>
    <parent link="RoboticARM_-_Link2-1_link"/>
    <child link="RoboticARM_-_Gripper-1_Gripper_base-1_link"/>
    <axis xyz="1.00000 0.00000 0.00000"/>
    <dynamics damping="0.7"/>
  </joint>

  <!-- Links -->
  <link name="RoboticARM_-_Base-2_Base_bottom-1_link">
    <collision name="RoboticARM - Base-2 Base_bottom-1_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00133 0.00000 0.03421"/>
      <geometry>
        <cylinder radius="0.82500" length="0.38500"/>
      </geometry>
    </collision>
    <!-- <collision name="RoboticARM - Base_Connector-1_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00133 -0.00000 0.47796"/>
      <geometry>
        <cylinder radius="0.25111" length="0.31000"/>
      </geometry>
    </collision> -->
    <inertial>
      <inertia ixx="1.58550" ixy="0.00012" ixz="0.05181" iyy="1.63580" iyz="-0.00010" izz="3.00072"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00367 0.00000 0.1299"/>
      <mass value="10.00000"/>
    </inertial>
    <visual name="RoboticARM - Base-2 Base_bottom-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 0.00000 0"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Base-2 Base_bottom-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="red"/>
    </visual>
    <visual name="RoboticARM - Base-2 Base_top-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00178 -0.00017 0.27240"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Base-2 Base_top-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="red"/>
    </visual>
    <visual name="RoboticARM - Base-2 straight miter gear_am-2">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.74310 0.00000 0.09562"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Base-2 straight miter gear_am-2.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="red"/>
    </visual>
    <visual name="RoboticARM - Base-2 straight miter gear_am-3">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00177 -0.00008 0.00747"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Base-2 straight miter gear_am-3.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="red"/>
    </visual>
    <visual name="RoboticARM - Base_Connector-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00177 -0.00000 0.47283"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Base_Connector-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>
  <link name="RoboticARM_-_Gripper-1_Gripper_base-1_link">
    <collision name="RoboticARM - Gripper-1 Gripper_base-1_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00125 -0.00021 0.40303"/>
      <geometry>
        <box size="0.66914 0.50137 0.47794"/>
      </geometry>
    </collision>
    <collision name="RoboticARM - Gripper-1 Gripper_clamp-1_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.19818 0.06984 1.14998"/>
      <geometry>
        <box size="0.21358 0.06043 0.60620"/>
      </geometry>
    </collision>
    <collision name="RoboticARM - Gripper-1 Gripper_clamp-2_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.18699 0.07031 1.14998"/>
      <geometry>
        <box size="0.21616 0.06043 0.60469"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00137" ixy="-0.00000" ixz="-0.00001" iyy="0.00154" iyz="0.00000" izz="0.00017"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00637 0.08715 1.09527"/>
      <mass value="0.02000"/>
    </inertial>
    <visual name="RoboticARM - Gripper-1 Gripper_arm-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.21716 0.08531 0.59226"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_arm-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 Gripper_arm-2">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.21093 0.08627 0.59483"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_arm-2.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 Gripper_base-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00497 -0.02716 0.0"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_base-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 Gripper_clamp-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.25486 0.08660 1.09314"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_clamp-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 Gripper_clamp-2">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.24212 0.08771 1.09740"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_clamp-2.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 Gripper_link-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.21916 0.06270 0.88928"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_link-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 Gripper_link-2">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.22816 0.06170 0.88634"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 Gripper_link-2.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 spur gear_am-2">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.14793 0.02513 0.50026"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 spur gear_am-2.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 spur gear_am-3">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.15181 0.02446 0.49910"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 spur gear_am-3.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Gripper-1 spur gear_am-4">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.06345 0.02505 0.26512"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Gripper-1 spur gear_am-4.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="green"/>
    </visual>
    <visual name="RoboticARM - Joint-2">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00001 -0.00092 -0.00089"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Joint-2.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="orange"/>
    </visual>
  </link>
  <link name="RoboticARM_-_Link1-1_link">
    <collision name="RoboticARM - Link1-1_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00078 0.00567 0.55762"/>
      <geometry>
        <cylinder radius="0.21651" length="0.90610"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.01074" ixy="0.00000" ixz="-0.00001" iyy="0.01154" iyz="-0.00010" izz="0.00259"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00080 0.00631 0.57335"/>
      <mass value="0.10000"/>
    </inertial>
    <visual name="RoboticARM - Joint-3">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00000 -0.00000 0.00000"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Joint-3.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <visual name="RoboticARM - Link1-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="0.00074 0.00586 0"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Link1-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <link name="RoboticARM_-_Link2-1_link">
    <collision name="RoboticARM - Link2-1_collision">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00044 0.00129 0.47416"/>
      <geometry>
        <cylinder radius="0.25091" length="0.54675"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="0.00638" ixy="0.00000" ixz="0.00001" iyy="0.00755" iyz="-0.00001" izz="0.00291"/>
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00109 0.00116 0.42112"/>
      <mass value="0.10000"/>
    </inertial>
    <visual name="RoboticARM - Joint-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00000 -0.00000 -0.00000"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Joint-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <visual name="RoboticARM - Link2-1">
      <origin rpy="0.00000 0.00000 0.00000" xyz="-0.00115 0.00123 0.0"/>
      <geometry>
        <mesh filename="package://sim/meshes/stl/RoboticARM - Link2-1.stl" scale="1.00000 1.00000 1.00000"/>
      </geometry>
      <material name="yellow"/>
    </visual>
  </link>
  
  <!-- Tramission -->
  <transmission name="trans_b_joint">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="b_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_b">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans_m_joint">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="m_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_m">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="trans_t_joint">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="t_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_t">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

</robot>